napari/layers/labels/labels.py

```diff
@@ -312,23 +307,33 @@ def mode(self, mode: Union[str, Mode]):
         if mode == self._mode:
             return
 
+        if self._mode == Mode.PICK:
+            self.mouse_drag_callbacks.remove(pick)
+        elif self._mode == Mode.PAINT:
+            self.mouse_drag_callbacks.remove(paint)
+        elif self._mode == Mode.FILL:
+            self.mouse_drag_callbacks.remove(fill)
```

Comment:
This feels brittle... Could we not have a stack-type API, so that we could just use `self.mouse_drag_callbacks.pop()` regardless of what is currently active?

Reply:
This will also change in #1048, but I think this matches our current API and approach and this can be fixed then